Add Context menu with Pin option to LHN

Gonals opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
Surrounding the existing TouchableOpacity with a PressableWithSecondaryInteraction component did not work on mobile, as the triggered touch event was the TouchableOpacity one, so I replaced the TouchableOpacity with the PressableWithSecondaryInteraction and gave it opacity capabilities. All working nice now!

### Fixed Issues
<!---
1. Please replace GH_LINK with a URL link to the GitHub issue this Pull Request is fixing.
2. Please replace PROPOSAL: GH_LINK_ISSUE(COMMENT) with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>
$ https://github.com/Expensify/App/issues/<number-of-the-issue(comment)>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/App/issues/15960


### Tests
1. Log into NewDot
2. On the LHN, right click/long press on top of an unpinned chat.
3. A menu with the pin option should show up. Press it
4. Confirm it gets pinned
5. Do the same for a pinned chat and confirm you can unpin it
6. Open newdot in a different devide and log into the same account.
7. On Devide 1  right click/long press on top of a chat to display the menu. Leave it open
8. On Devide 2, pin/unpin the chat for which you open the menu on Device 1.
9. When the pin status update id Device 1, confirm the menu is automatically closed.
10. Open a chat and right click/long press one of the messages. Confirm the usual menu shows up as usual and it doesn't have the pin/unpin option.
11. On that menu, select the "Delete" option and confirm the message is correctly deleted
12. (On Web/Desktop only) Go to Settings > About > App Download links and right click on one of the links. A menu should show up with just the option to copy the link.

- [x] Verify that no errors appear in the JS console

### Offline tests
Not relevant

### QA Steps
Same as tests

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<img width="1443" alt="Screenshot 2023-03-17 at 4 01 30 PM" src="https://user-images.githubusercontent.com/19366280/226022558-c7c8a859-0d1a-4e5e-8bcd-da66878eca24.png">


</details>

<details>
<summary>Mobile Web - Chrome</summary>

<img width="384" alt="Screenshot 2023-03-22 at 4 39 00 PM" src="https://user-images.githubusercontent.com/19366280/226957807-02a7e823-27db-49b4-b051-eff9e984ee50.png">


</details>

<details>
<summary>Mobile Web - Safari</summary>

<img width="425" alt="Screenshot 2023-03-22 at 4 34 09 PM" src="https://user-images.githubusercontent.com/19366280/226957864-1e4fc6b9-0af3-48ae-908a-d5b86f3ddebb.png">


</details>

<details>
<summary>Desktop</summary>

<img width="1196" alt="Screenshot 2023-03-17 at 4 02 43 PM" src="https://user-images.githubusercontent.com/19366280/226022780-3684dcac-e4a9-4d73-b3fe-1aaca069dacc.png">


</details>

<details>
<summary>iOS</summary>

<img width="418" alt="Screenshot 2023-03-22 at 4 33 46 PM" src="https://user-images.githubusercontent.com/19366280/226957926-cd4d3d6a-7d71-4fb9-b192-8b7ac316e33c.png">


</details>

<details>
<summary>Android</summary>

<img width="385" alt="Screenshot 2023-03-22 at 4 38 10 PM" src="https://user-images.githubusercontent.com/19366280/226957972-a8a262b7-ab8c-44cd-8ed8-0922dd00ac32.png">


</details>

Gonals committed:
changing branchs

Gonals committed:
Merge branch 'main' into alberto-pin

Gonals committed:
add PIN action

Gonals committed:
correctly show menu

Gonals committed:
handle pin/unpin behavior

Gonals committed:
no need fur success messages

Gonals committed:
figure out what's going on

Gonals committed:
make pin/unpin work fine

Gonals committed:
cleanup

Gonals committed:
lint

Gonals committed:
more lint cleanup

Gonals committed:
update test

Gonals committed:
handle mobile devices

Gonals committed:
Merge branch 'main' into alberto-pin

Gonals committed:
Use pressablewithSecondaryInteraction instead of touchableOpacity

Gonals committed:
Turn Pressable into touchableOpacity

Gonals committed:
cleanUp

Gonals committed:
update test

Gonals commented:
OFF WIP! All yours @cead22

cead22 commented:
I revived [this slack discussion ](https://expensify.slack.com/archives/CC7NECV4L/p1679597868279769?thread_ts=1678740215.629059&cid=CC7NECV4L) because I'm not convinced this is a good change

cead22 commented on src/components/LHNOptionsList/OptionRowLHN.js:
```diff
@@ -98,6 +100,28 @@ const OptionRowLHN = (props) => {
+            props.reportID,
+            {},
+            '',
+            undefined,
+            undefined,
```
Should these be `() => {}` since that's their default value?

cead22 reviewed (commented):
I think we this PR should have the following tests, but I'm curious for your thoughts
- All uses of showContextMenu
- All uses of PressableWithSecondaryInteraction
- Deleting a report and make sure nothing breaks since hideDeleteModal changed

Gonals committed:
Merge branch 'main' into alberto-pin

Gonals committed:
add better defaults

Gonals commented:
> I think we this PR should have the following tests, but I'm curious for your thoughts
> 
> * All uses of showContextMenu

These changes shouldn't have affected other menus, right? We can add a test to just make sure the new "Pin/Unpin" doesn't show for other menus, but I think it'd be enough to change in `any` other use of the menu, instead of all of them

> * All uses of PressableWithSecondaryInteraction

Yep. This was a bigger one. Adding tests.

> * Deleting a report and make sure nothing breaks since hideDeleteModal changed

👍

shawnborton commented:
There should be top/bottom padding above the single menu option here. That will help with legibility:
<img width="378" alt="image" src="https://user-images.githubusercontent.com/2319350/229909172-3f13c1aa-383b-41a7-b326-722bcb1a92e8.png">


Actually I'm not sure how/when/why but it looks like that padding got removed from a bunch of places, when it was intended and not a bug. @kavimuru do you recall anywhere that might have been reported by mistake?

Gonals commented on src/components/PressableWithSecondaryInteraction/index.js:
```diff
@@ -53,7 +53,7 @@ class PressableWithSecondaryInteraction extends Component {
         // On Web, Text does not support LongPress events thus manage inline mode with styling instead of using Text.
         return (
-            <Pressable
+            <TouchableOpacity
```
We need this to be a `TouchableOpacity` for the ReportActions. It is not a "new" one, we are just moving it around. Once we decide how we want to handle those, then we'll replace it here.

cead22 replied:
@0xmiroslav does this blurb from the PR description answer your question?

>Surrounding the existing TouchableOpacity with a PressableWithSecondaryInteraction component did not work on mobile, as the triggered touch event was the TouchableOpacity one, so I replaced the TouchableOpacity with the PressableWithSecondaryInteraction and gave it opacity capabilities. All working nice now!

Gonals replied:
ReportActions currently use a TouchableOpacity, and the current UI is based on that. I'm moving that TouchableOpacity to the PressableWithSecondaryInteraction component (which currently surrounds the TouchableOpacity for ReportActions) so that it works correctly in all cases. Using Pressable here would mess the UI.

We are currently designing/developing a new component to replace TouchableOpacity. When that happens, it'll simply get replaced here too.

Gonals committed:
Merge branch 'main' into alberto-pin

Gonals commented on src/components/LHNOptionsList/OptionRowLHN.js:
```diff
@@ -105,15 +129,19 @@ const OptionRowLHN = (props) => {
         >
             <Hoverable>
```
We have `hovered` available for styling purposes. I think that's the main reason I have it this way

Gonals committed:
remove unneeded props

Gonals commented:
Answered a few comments!

cead22 commented:
>I think it'd be enough to change in any other use of the menu, instead of all of them

Tbh I think that's reasonable. I'm just trying to avoid any regressions, since I've approved some PRs before that created non-obvious ones. So this is reasonable, but testing all of them would be better


In [src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js](https://github.com/Expensify/App/pull/16079/files#diff-16a590a45460a9abf1e402441b50e3cb2659ee3da203ca30a2fa4ce3f9411957R271)
<img width="1111" alt="image" src="https://user-images.githubusercontent.com/165133/230453024-cd1cb4f1-4f6f-4f61-b33e-e75174b1c803.png">

Gonals committed:
Merge branch 'main' into alberto-pin

Gonals commented:
> > I think it'd be enough to change in any other use of the menu, instead of all of them
> 
> Tbh I think that's reasonable. I'm just trying to avoid any regressions, since I've approved some PRs before that created non-obvious ones. So this is reasonable, but testing all of them would be better

There's only two of them, so sure 😆. Added! 
 
> > Where did it change? 😕
> 
> In [src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js](https://github.com/Expensify/App/pull/16079/files#diff-16a590a45460a9abf1e402441b50e3cb2659ee3da203ca30a2fa4ce3f9411957R271) <img alt="image" width="1111" src="https://user-images.githubusercontent.com/165133/230453024-cd1cb4f1-4f6f-4f61-b33e-e75174b1c803.png">

Ah! That's deleting a comment 😁. Added to the instructions.

Gonals commented:
Addressed comments!

cead22 commented:
It looks like a bug to me, and when I right click on a message on web, and then click outside of it, the menu is hidden. I think that behavior should be the same here

Gonals commented:
Hey guys! I'm going on parental leave as my daughter was born early! @cead22, feel free to either wrap this up or just wait for me to come back (I'll only be gone for a month), since it is hardly an urgent issue!

Gonals committed:
conflicts

Gonals committed:
Merge branch 'main' into alberto-pin

Gonals committed:
more conflicts

Gonals committed:
back to pressable

Gonals committed:
pass params better

Gonals commented:
I'm back! It took a bit of updating, but this is now working again!

Gonals commented:
> It looks like a bug to me, and when I right click on a message on web, and then click outside of it, the menu is hidden. I think that behavior should be the same here

What do you mean? That seems like a completely different case to me. I think what's going on here is:
1. The "Pin" menu is opened in Web
2. The user unpins/pins the chat from mobile/desktop
3. The menu remains open in web.

I think that is fine, and handling closing the menu automatically in this cases doesn't make a lot of sense (similarly, the menu is not automatically closed if a user edits a comment from a different device, for example)

cead22 commented:
> What do you mean?

It's been a minute but looking at the video I think I missed the phone before, and thought that click to pin was happening on the web app. So I agree with what you said is what's going on

I was just sharing what I expected would happen, thinking about it from a UX perspective. Another thing I noticed this time around is that clicking the lingering Pin button, didn't pin the concierge conversation. That's another thing I expected to happen that doesn't look like it happened

Gonals commented:
> > What do you mean?
> 
> It's been a minute but looking at the video I think I missed the phone before, and thought that click to pin was happening on the web app. So I agree with what you said is what's going on

Ah, gotcha! 


> I was just sharing what I expected would happen, thinking about it from a UX perspective. 

So you think it'd be worth it to try to hide the menu if the Pin situation changes? Note that we'll be adding the "call" button here too (and potentially other things).


> Another thing I noticed this time around is that clicking the lingering Pin button, didn't pin the concierge conversation. That's another thing I expected to happen that doesn't look like it happened

That's because it was the "Unpin" button and the conversation was already unpinned 😛. Had it been pinned, it would have unpinned it.
The main reason behind it is that we don't really have a good way to change the text of the menu action based on a parameter (ispinned being true/false), so `pin` and `unpin` are two separate actions.

Gonals committed:
lint

Gonals committed:
prettier

Gonals committed:
conflicts

cead22 commented:
> So you think it'd be worth it to try to hide the menu if the Pin situation changes? Note that we'll be adding the "call" button here too (and potentially other things).

Looking at the video again, since the conversation is moving, I think it would be good to hide the context menu, since the "context" is no longer there

> That's because it was the "Unpin" button and the conversation was already unpinned 😛. Had it been pinned, it would have unpinned it. The main reason behind it is that we don't really have a good way to change the text of the menu action based on a parameter (ispinned being true/false), so `pin` and `unpin` are two separate actions.

🤦‍♂️ that makes sense, I didn't think of that. I think hiding the menu sounds better than updating the text, cause then the text would be updated but the context menu would be over a different conversation cause the original one may have moved

Gonals committed:
Merge branch 'main' into alberto-pin

Gonals committed:
close popup if pin changes

Gonals committed:
remove unneeded import

cead22 commented:
Sorry for the delay, I was focused on getting [this PR](https://github.com/Expensify/Expensify/issues/265968) ready for re-review. I'll review this again tomorrow!

cead22 commented on src/pages/home/report/ContextMenu/ContextMenuActions.js:
```diff
@@ -264,6 +265,30 @@ export default [
+    {
+        textTranslateKey: 'common.pin',
+        icon: Expensicons.Pin,
+        shouldShow: (type, reportAction, isArchivedRoom, betas, anchor, isChronosReport, reportID, isPinnedChat) => type === CONTEXT_MENU_TYPES.REPORT && !isPinnedChat,
+        onPress: (closePopover, {reportID}) => {
+            Report.togglePinnedState(reportID, false);
+            if (closePopover) {
+                hideContextMenu(false);
+            }
+        },
```
Is it worth it to DRY this?

Gonals replied:
You mean avoiding having two for pin/unpin? I tried a few options and even asked other engineers during ECX. Apparently, this is the way 🤷

cead22 replied:
Yeah I noticed shouldShow is the same for both, and onPress is similar but with the opposite bool values, but I'm good with this as is

cead22 commented on src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js:
```diff
@@ -217,6 +232,28 @@ class PopoverReportActionContextMenu extends React.Component {
         });
     }
 
+    /**
+     * Used to calculate the Context Menu Dimensions
+     *
+     * @returns {JSX}
+     */
+    measureContent() {
+        return (
+            <BaseReportActionContextMenu
+                type={this.state.type}
+                isVisible
+                selection={this.state.selection}
+                reportID={this.state.reportID}
+                reportAction={this.state.reportAction}
+                isArchivedRoom={this.state.isArchivedRoom}
+                isChronosReport={this.state.isChronosReport}
+                isPinnedChat={this.state.isPinnedChat}
+                anchor={this.contextMenuTargetNode}
+                contentRef={this.setContentRef}
+            />
+        );
+    }
+
```
```suggestion
```

This looks like leftover from resolving conflicts when merging main, since I see that was removed [here](https://github.com/Expensify/App/pull/18883/files#diff-16a590a45460a9abf1e402441b50e3cb2659ee3da203ca30a2fa4ce3f9411957L47)

cead22 commented:
@Gonals code looks good, thanks for the updated tests to check that the context menu is hidden if the pinned state change
@0xmiroslav can you complete the reviewer checklist when you get a chance please?

Gonals committed:
Merge branch 'main' into alberto-pin

Gonals committed:
remove measure

Gonals committed:
Merge branch 'main' into alberto-pin

Gonals commented on src/components/PressableWithSecondaryInteraction/index.native.js:
```diff
@@ -1,6 +1,6 @@
 import _ from 'underscore';
 import React, {forwardRef} from 'react';
-import {Pressable} from 'react-native';
+import {TouchableOpacity} from 'react-native';
```
Ah, yes, I changed this back in index.js, but forgot to do so for native. That's also what's causing the error.

Gonals committed:
use pressable

Gonals commented:
Comments addressed! It should work now on native!

cead22 reviewed (dismissed):
@0xmiroslav please re-review & re-test when you get a chance

Gonals committed:
conflicts

cead22 commented:
@Gonals conflicts again 😬

Gonals committed:
conflicts

Gonals committed:
remove unused param

Gonals commented on src/pages/home/report/ContextMenu/ReportActionContextMenu.js:
```diff
@@ -30,12 +30,12 @@ function showContextMenu(
         return;
     }
-    contextMenuRef.current.showContextMenu(type, event, selection, contextMenuAnchor, reportID, reportAction, draftMessage, onShow, onHide, isArchivedRoom, isChronosReport, childReportID);
+    contextMenuRef.current.showContextMenu(type, event, selection, contextMenuAnchor, reportID, reportAction, draftMessage, onShow, onHide, isArchivedRoom, isChronosReport, isPinnedChat);
```
Done!

Gonals commented:
Conflicts solved AGAIN!

Gonals committed:
prettier

0xmiroslav reviewed (approved):
👍

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5153857375) to staging by https://github.com/cead22 in version: 1.3.23-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|cancelled 🔪 
🖥 desktop 🖥|failure ❌
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|cancelled 🔪

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5180754033) to production by https://github.com/roryabraham in version: 1.3.23-7 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|failure ❌
🍎 iOS 🍎|failure ❌ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5180754033) to production by https://github.com/roryabraham in version: 1.3.23-7 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|failure ❌
🍎 iOS 🍎|failure ❌ 
🕸 web 🕸|success ✅

roryabraham commented on src/components/LHNOptionsList/OptionRowLHN.js:
```diff
@@ -52,11 +55,16 @@ const defaultProps = {
 const OptionRowLHN = (props) => {
     const optionItem = SidebarUtils.getOptionData(props.reportID);
+
+    React.useEffect(() => {
```
I'm actually not seeing why this is necessary. I removed this and everything seems to work fine.

roryabraham commented:
I think that using `PopoverMenu` may have been a more appropriate component than `ReportActionContextMenu`, which is specifically for ... report actions and their context menus.

abdulrahuman5196 commented:
Minor miss in implementation - Pin/Unpin should have made the composer in focus - https://github.com/Expensify/App/issues/21861

sobitneupane commented:
[This case](https://github.com/Expensify/App/issues/21929) was not considered while implementing the context menu in Sidebar.

Issue: Context menu from LHN is visible even if report page is opened on resize

Steps to reproduce:

1. Right click an option in LHN.
2. Resize the window to small while the popover is still open.
